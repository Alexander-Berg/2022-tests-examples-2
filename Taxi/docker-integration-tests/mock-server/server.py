#!/usr/bin/env python3
import asyncio
import logging
import os
import socket

from aiohttp import web
import uvloop

from mock_server import application

PORTS = [80, 8018, 8025, 8031, 8038, 17000]
ACCESS_LOG_FORMAT = (
    '%{Host}i %a %l %u %t "%r" %s %b "%{Referrer}i" "%{User-Agent}i'
)
LOG_FILE_TPL = '/taxi/logs/application-%s.log'


def init_loggers():
    program_name = os.getenv('program_name')
    logger = logging.getLogger()
    handler = logging.FileHandler(LOG_FILE_TPL % program_name)
    handler.setLevel(logging.DEBUG)
    logger.setLevel(logging.DEBUG)
    logger.addHandler(handler)


def run():
    init_loggers()
    asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
    app = application.get_application()
    sockets = []
    for port in PORTS:
        sock = socket.socket(socket.AF_INET6)
        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, True)
        sock.bind(('::', port))
        sockets.append(sock)
    web.run_app(app, sock=sockets, access_log_format=ACCESS_LOG_FORMAT)


if __name__ == '__main__':
    run()
