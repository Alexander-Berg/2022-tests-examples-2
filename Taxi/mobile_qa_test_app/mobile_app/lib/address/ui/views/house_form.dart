import 'package:flutter_hooks/flutter_hooks.dart';
import 'package:flutter_txm_ui_components/components.dart';
import 'package:hooks_riverpod/hooks_riverpod.dart';
import '../../../utils/localization.dart';
import '../../address_providers.dart';

class HouseForm extends HookConsumerWidget {
  const HouseForm({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final pageManager = ref.watch(addressPageManager);
    final pageStateHolder = ref.watch(addressPageStateHolderProvider.notifier);
    final address = pageStateHolder.address;
    final houseInputController = useTextEditingController(text: address.house);
    final corpusInputController =
        useTextEditingController(text: address.corpus);
    final buildingInputController =
        useTextEditingController(text: address.building);

    return Row(
      children: [
        Expanded(
          child: YXListInput(
            controller: houseInputController,
            borderType: YXListBorderType.bottom,
            placeholderHasDots: false,
            placeholder: Strings.of(context).houseInputPlaceholder,
            onChanged: pageManager.onChangedHouse,
          ),
        ),
        Expanded(
          child: YXListInput(
            controller: corpusInputController,
            borderType: YXListBorderType.bottom,
            placeholderHasDots: false,
            placeholder: Strings.of(context).corpusInputPlaceholder,
            onChanged: pageManager.onChangedCorpus,
          ),
        ),
        Expanded(
          child: YXListInput(
            controller: buildingInputController,
            borderType: YXListBorderType.bottom,
            placeholderHasDots: false,
            placeholder: Strings.of(context).buildingInputPlaceholder,
            onChanged: pageManager.onChangedBuilding,
          ),
        ),
      ],
    );
  }
}
