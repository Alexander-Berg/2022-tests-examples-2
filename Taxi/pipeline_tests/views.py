import copy
import typing

from aiohttp import web

from js_pipeline.generated import models

from admin_pipeline import common
from admin_pipeline import common_helpers
from admin_pipeline.generated.service.swagger.models import api


def drop_id(doc: dict) -> dict:
    data = copy.copy(doc)
    if '_id' in doc:
        del data['_id']
    if 'id' in doc:
        del data['id']
    return data


class NotFoundError(Exception):
    ERR_CODE = 'not_found'


class _CommonView(common_helpers.DBHelper):
    DATA_TYPE: typing.Optional[typing.Type] = None

    @staticmethod
    def __for_db(data):
        return common.for_db(data, id_type=str)

    @classmethod
    async def create(cls, request, context):
        data = request.body
        existing = await cls.try_find_one(
            context.mongo.secondary,
            {'_id': data.id},
            request.consumer,
            for_model=cls.DATA_TYPE,
        )
        if existing is not None:
            raise web.HTTPBadRequest(
                reason=f'Item with id ({data.id}) already exists',
            )
        serialized = cls.__for_db(cls.DATA_TYPE.serialize(data))
        serialized['consumer'] = request.consumer
        await cls.insert_one(context.mongo, serialized)

    @classmethod
    async def get(cls, id_, consumer, context):
        found = await cls.try_find_one(
            context.mongo.secondary,
            {'_id': id_},
            consumer,
            for_model=cls.DATA_TYPE,
        )
        if not found:
            raise NotFoundError('Item not found: %s for %s' % (id_, consumer))
        return cls.DATA_TYPE.deserialize(common.for_json(found))

    @classmethod
    async def update_item(cls, request, context):
        item = request.body
        if item.id != request.id:
            raise web.HTTPBadRequest(
                reason='Item id changed! Old %s, new - %s '
                % (request.id, item.id),
            )
        doc = cls.__for_db(cls.DATA_TYPE.serialize(item))
        doc['consumer'] = request.consumer
        result = await cls.update(
            context.mongo,
            {'$and': [{'_id': request.id}, {'consumer': request.consumer}]},
            doc,
        )
        if result['n'] != 1:
            raise NotFoundError(
                'Item not found: %s for %s' % (request.id, request.consumer),
            )

    @classmethod
    async def delete(cls, request, context):
        await cls.remove(
            context.mongo,
            {'$and': [{'_id': request.id}, {'consumer': request.consumer}]},
        )


class ChecksView(_CommonView):
    DATA_TYPE = api.Check
    PROJECTIONS = {DATA_TYPE: ['name', 'check']}

    @classmethod
    def determine_check_type(cls, check):
        if 'source_code' in check:
            return 'imperative'
        if 'expected_output' in check:
            return 'combined'
        raise Exception('Unexpected check type: ' + str(check))

    @classmethod
    def get_collection(cls, db):
        return db.admin_pipeline_test_checks

    @classmethod
    async def enumerate_checks(cls, consumer, context):
        all_checks = cls.find_many(
            context.mongo, {}, consumer, for_model=api.Check,
        )
        result = api.TestChecksEnumerateResponse(
            output_checks=[
                api.EnumeratedCheck(
                    check['_id'],
                    check['name'],
                    cls.determine_check_type(check['check']),
                )
                async for check in all_checks
            ],
        )
        return result


class MocksView(_CommonView):
    DATA_TYPE = api.Mock
    PROJECTIONS = {DATA_TYPE: ['name', 'resource', 'is_prefetched', 'mock']}

    @classmethod
    def get_collection(cls, db):
        return db.admin_pipeline_test_mocks

    @classmethod
    async def enumerate_mocks(cls, consumer, context):
        all_mocks = cls.find_many(
            context.mongo, {}, consumer, for_model=api.Mock,
        )
        input_mocks = []
        prefetched_resources_mocks = []
        resources_mocks = []
        async for mock in all_mocks:
            resource = mock.get('resource')
            name = mock['name']
            _id = mock['_id']
            is_prefetched = mock['is_prefetched']
            if not resource:
                input_mocks.append(api.EnumeratedMocksItem(_id, name))
            elif is_prefetched:
                prefetched_resources_mocks.append(
                    api.EnumeratedMocksItem(_id, name, resource),
                )
            else:
                resources_mocks.append(
                    api.EnumeratedMocksItem(_id, name, resource),
                )

        result = api.TestMocksEnumerateResponse(
            input_mocks=input_mocks,
            resources_mocks=resources_mocks,
            prefetched_resources_mocks=prefetched_resources_mocks,
        )
        return result


class TestsView(_CommonView):
    DATA_TYPE = models.PipelineTest
    PROJECTIONS = {
        DATA_TYPE: [
            'name',
            'scope',
            'testcases',
            'prefetched_resources_mocks',
            'resources_mocks',
            'input_mocks',
            'output_checks',
        ],
    }

    @classmethod
    def get_collection(cls, db):
        return db.admin_pipeline_tests

    @classmethod
    async def enumerate_tests(cls, request, context):
        all_tests = cls.find_many(
            context.mongo, {}, request.consumer, for_model=models.PipelineTest,
        )
        result = api.TestsEnumerateResponse(
            tests=[
                api.EnumeratedTest(test['_id'], test['name'], test['scope'])
                async for test in all_tests
            ],
        )
        return result


class ChoosenTestsHelper(common_helpers.DBHelper):
    PROJECTIONS: dict = dict()

    @classmethod
    def get_collection(cls, db):
        return db.admin_pipeline_choosen_tests


class TestsResultsHelper(common_helpers.DBHelper):
    PROJECTIONS: dict = {models.PipelineTestsResults: ['tests', 'created']}

    @classmethod
    def get_collection(cls, db):
        return db.admin_pipeline_tests_results
