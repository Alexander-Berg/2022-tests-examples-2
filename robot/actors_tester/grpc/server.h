#pragma once

#include <robot/zora/tools/actors_tester/grpc/proto/proto.pb.h>
#include <robot/zora/tools/actors_tester/grpc/proto/proto.grpc.pb.h>

#include <contrib/libs/grpc/include/grpc++/grpc++.h>

#include <util/generic/ptr.h>
#include <util/system/thread.h>
#include <util/datetime/base.h>

class TGrpcServer final {
public:
    using TCallback = std::function<void(NAsyncGrpcTest::Request request)>;
    TGrpcServer(TString uri);
    ~TGrpcServer();

    void Start();
    void SetCallback(TCallback cb); //TODO: need to associate types of requests to callbacks

private:

    class CallData {
    public:
        CallData(NAsyncGrpcTest::TestService::AsyncService* service, grpc::ServerCompletionQueue* cq, TCallback& callback);

        void Proceed();

    private:
        NAsyncGrpcTest::TestService::AsyncService* Service;
        grpc::ServerCompletionQueue* Cq;
        grpc::ServerContext Ctx;

        NAsyncGrpcTest::Request Request;
        NAsyncGrpcTest::Response Response;

        grpc::ServerAsyncResponseWriter<NAsyncGrpcTest::Response> Responder;

        enum CallStatus { CREATE, PROCESS, FINISH };
        CallStatus Status;
        TCallback& Callback;
    };

     void HandleRpcs();

    std::unique_ptr<grpc::ServerCompletionQueue> Cq;
    NAsyncGrpcTest::TestService::AsyncService Service;
    std::unique_ptr<grpc::Server> Server;

    TSimpleSharedPtr<TThread> Thread;
    TString Uri;
    TCallback Callback;
};
