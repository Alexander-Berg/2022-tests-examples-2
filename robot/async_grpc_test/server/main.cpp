#include <robot/zora/tools/async_grpc_test/proto/proto.pb.h>
#include <robot/zora/tools/async_grpc_test/proto/proto.grpc.pb.h>

#include <contrib/libs/grpc/include/grpc++/grpc++.h>

#include <util/generic/ptr.h>


class ServerImpl final {
public:
    ~ServerImpl() {
        Server->Shutdown();
        Cq->Shutdown();
    }
    
    void Run() {
        std::string server_address("0.0.0.0:50051");
        
        grpc::ServerBuilder builder;
        builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
        builder.RegisterService(&Service);
        Cq = builder.AddCompletionQueue();
        Server = builder.BuildAndStart();
        Cerr << "Server listening on " << server_address << Endl;
        
        // Proceed to the server's main loop.
        HandleRpcs();
    }
    
private:
    class CallData {
    public:
        CallData(NGrpcTest::TestService::AsyncService* service, grpc::ServerCompletionQueue* cq)
            : Service(service), Cq(cq), Responder(&Ctx), Status(CREATE) {
            Proceed();
        }

        void Proceed() {
            if (Status == CREATE) {
                Status = PROCESS;
                Service->RequestGetResponse(&Ctx, &Request, &Responder, Cq, Cq, this);
                Cerr << "Create" << Endl;

            } else if (Status == PROCESS) {
                new CallData(Service, Cq);

                TString prefix("Hello ");
                Response.SetSomeData(prefix + Request.GetSomeData());

                Status = FINISH;
                Cerr << "Process" << Endl;
                Responder.Finish(Response, grpc::Status::OK, this);

            } else {
                assert(Status == FINISH);
                delete this;
            }
        }

    private:
        NGrpcTest::TestService::AsyncService* Service;
        grpc::ServerCompletionQueue* Cq;
        grpc::ServerContext Ctx;

        NGrpcTest::Request Request;
        NGrpcTest::Response Response;

        grpc::ServerAsyncResponseWriter<NGrpcTest::Response> Responder;

        enum CallStatus { CREATE, PROCESS, FINISH };
        CallStatus Status;
    };

     void HandleRpcs() {
       new CallData(&Service, Cq.get());
       void* tag;  // uniquely identifies a request.
       bool ok;
       while (true) {
         assert(Cq->Next(&tag, &ok));
         assert(ok);
         static_cast<CallData*>(tag)->Proceed();
       }
     }
    
    std::unique_ptr<grpc::ServerCompletionQueue> Cq;
    NGrpcTest::TestService::AsyncService Service;
    std::unique_ptr<grpc::Server> Server;
};


int main(int, char**) {
    ServerImpl server;
    server.Run();

    return 0;
}
