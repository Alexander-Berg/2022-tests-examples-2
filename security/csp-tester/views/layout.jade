doctype html
html
  head
    title= title
    link(rel='stylesheet', href='/stylesheets/style.css')
    link(rel='stylesheet', href='/stylesheets/highlight/styles/tomorrow.css')
    script(src='/javascripts/highlight.pack.js')
  body
    h1 CSP Tester
    script hljs.initHighlightingOnLoad();
    block content
