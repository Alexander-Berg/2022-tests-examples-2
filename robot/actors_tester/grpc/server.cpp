#include "server.h"


TGrpcServer::TGrpcServer(TString uri) : Uri(uri) {}

TGrpcServer::~TGrpcServer() {
    Server->Shutdown();
    Cq->Shutdown();
}

void TGrpcServer::Start() {
    grpc::ServerBuilder builder;
    builder.AddListeningPort(Uri, grpc::InsecureServerCredentials());
    builder.RegisterService(&Service);
    Cq = builder.AddCompletionQueue();
    Server = builder.BuildAndStart();
    Cerr << "Server listening on " << Uri << Endl;

    // Proceed to the server's main loop.
    Thread.Reset(MakeSimpleShared<TThread>([this](){
        HandleRpcs();
    }));
}

void TGrpcServer::SetCallback(TGrpcServer::TCallback cb)
{
    Callback = cb;
}

void TGrpcServer::HandleRpcs() {
    new CallData(&Service, Cq.get(), Callback);
    void* tag = nullptr;  // uniquely identifies a request.
    bool ok = false;
    while (ok) {
        assert(Cq->Next(&tag, &ok));
        static_cast<CallData*>(tag)->Proceed();
    }
}

TGrpcServer::CallData::CallData(NAsyncGrpcTest::TestService::AsyncService *service, grpc::ServerCompletionQueue *cq, TCallback &callback)
    : Service(service), Cq(cq), Responder(&Ctx), Status(CREATE), Callback(callback) {
    Proceed();
}

void TGrpcServer::CallData::Proceed() {
    if (Status == CREATE) {
        Status = PROCESS;
        Service->RequestGetResponse(&Ctx, &Request, &Responder, Cq, Cq, this);

    } else if (Status == PROCESS) {
        new CallData(Service, Cq, Callback);

        TString prefix("Hello ");
        Response.SetSomeData(prefix + Request.GetSomeData());

        Status = FINISH;
        Responder.Finish(Response, grpc::Status::OK, this);
//        try { //TODO
        Callback(Request);
//        } catch (...) {
//        }
    } else {
        assert(Status == FINISH);
        delete this;
    }
}
