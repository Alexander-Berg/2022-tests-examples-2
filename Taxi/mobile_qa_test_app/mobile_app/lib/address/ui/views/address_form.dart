import 'package:flutter_hooks/flutter_hooks.dart';
import 'package:flutter_txm_ui_components/components.dart';
import 'package:hooks_riverpod/hooks_riverpod.dart';

import '../../../auth/domain/user_state_holder.dart';
import '../../../utils/localization.dart';
import '../../address_providers.dart';
import 'house_form.dart';

class AddressForm extends HookConsumerWidget {
  const AddressForm({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final pageManager = ref.watch(addressPageManager);
    final address = ref.watch(userStateProvider.select((u) => u?.address));
    final cityInputController = useTextEditingController(text: address?.city);
    final streetInputController =
        useTextEditingController(text: address?.street);

    return Column(
      children: [
        YXListInput(
          controller: cityInputController,
          borderType: YXListBorderType.bottom,
          placeholderHasDots: false,
          placeholder: Strings.of(context).cityInputPlaceholder,
          onChanged: pageManager.onChangedCity,
        ),
        YXListInput(
          controller: streetInputController,
          borderType: YXListBorderType.bottom,
          placeholderHasDots: false,
          placeholder: Strings.of(context).streetInputPlaceholder,
          onChanged: pageManager.onChangedStreet,
        ),
        const HouseForm(),
      ],
    );
  }
}
