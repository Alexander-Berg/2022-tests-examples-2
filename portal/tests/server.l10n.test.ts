import { mockReq } from '@lib/views/mockReq';
import { logError } from '@lib/log/logError';
import { l10n } from '../server.l10n';

jest.mock('../../log/logError.ts', () => {
    return {
        logError: jest.fn().mockImplementation(() => {
            throw new Error('logError called!');
        })
    };
});
const mockedLogError = logError as jest.Mock;

describe('home.lang', function() {
    afterEach(() => {
        jest.clearAllMocks();
    });

    test('should return string', function() {
        expect(l10n('yandex')).toEqual('Яндекс');
        expect(l10n('mail.title')).toEqual('Почта');
    });

    test('should return array', function() {
        expect(l10n('mail.num.mail')).toEqual(['письмо', 'письма', 'писем']);
    });

    test('should return empty string if not found', function() {
        mockedLogError.mockImplementation(() => {});

        expect(l10n('some.inexistent.thing')).toEqual('');
        expect(l10n('mail.some.inexistent.thing')).toEqual('');
        expect(l10n('__proto__')).toEqual('');
        expect(l10n('mail.__proto__')).toEqual('');

        expect(mockedLogError.mock.calls).toMatchSnapshot();
    });

    test('should work with different locales', function() {
        expect(l10n('mail.title', 'ua')).toEqual('Пошта');
        expect(l10n.call(mockReq({}, { Locale: 'ua' }), 'mail.title')).toEqual('Пошта');
    });
});
