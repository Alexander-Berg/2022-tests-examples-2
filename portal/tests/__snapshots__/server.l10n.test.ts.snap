// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home.lang should return empty string if not found 1`] = `
Array [
  Array [
    Object {
      "block": "server.l10n",
      "level": "warn",
      "message": "undefined translation some.inexistent.thing in lang ru",
    },
  ],
  Array [
    Object {
      "block": "server.l10n",
      "level": "warn",
      "message": "undefined translation mail.some.inexistent.thing in lang ru",
    },
  ],
  Array [
    Object {
      "block": "server.l10n",
      "message": "__proto__ usage",
    },
  ],
  Array [
    Object {
      "block": "server.l10n",
      "message": "__proto__ usage",
    },
  ],
]
`;
